presto-spark-base/src/test/java/com/facebook/presto/spark/task/TestNonNativePrestoSparkTaskExecution.java

```diff
@@ -0,0 +1,107 @@
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+
+public class TestNonNativePrestoSparkTaskExecution
+{
+    Session nonNativeTestSession;
+    LocalExecutionPlanner.LocalExecutionPlan localExecutionPlan;
+    TaskExecutor taskExecutor = new TaskExecutor(8, 16, 3, 4, TASK_FAIR, Ticker.systemTicker());
+    PlanFragment planFragment = createPlanFragment();
+    ExecutorService taskNotificationExecutor = newCachedThreadPool(daemonThreadsNamed("test-executor-%s"));
+    ScheduledExecutorService scheduledExecutor = newScheduledThreadPool(2, daemonThreadsNamed("test-scheduledExecutor-%s"));
+    ScheduledSplit split1 = new ScheduledSplit(1, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit()));
+    ScheduledSplit split2 = new ScheduledSplit(2, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit()));
+    ScheduledSplit split3 = new ScheduledSplit(3, TABLE_SCAN_NODE_ID, new Split(new ConnectorId("test"), TestingTransactionHandle.create(), TestingSplit.createLocalSplit()));
+    TaskStateMachine taskStateMachine;
+
+    @BeforeClass
+    public void setUp()
+    {
+        nonNativeTestSession = testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema(TINY_SCHEMA_NAME)
+                .build();
+        taskStateMachine = new TaskStateMachine(new TaskId("test_query_id", 4, 4, 1), taskNotificationExecutor);
+
+        localExecutionPlan = createTestingPlanner().plan(
+                TestingTaskContext.createTaskContext(taskNotificationExecutor, scheduledExecutor, nonNativeTestSession),
+                planFragment,
+                new UnsupportedOutputBuffer(),
+                new UnsupportedRemoteSourceFactory(),
+                new TableWriteInfo(Optional.empty(), Optional.empty(), Optional.empty()),
+                new ArrayList<>());
+    }
+
+    @AfterClass
+    public void afterClass()
+    {
+        taskStateMachine.finished();
+        taskExecutor.stop();
+        scheduledExecutor.shutdown();
+        taskNotificationExecutor.shutdown();
+    }
+
+    @Test
+    public void testNonNativeSourceNodeDriverSplitRunnerInstanceCount()
+    {
+        TaskContext nonNativeTaskContext = TestingTaskContext.createTaskContext(taskNotificationExecutor, scheduledExecutor, nonNativeTestSession, new DataSize(2, GIGABYTE));
+        taskExecutor.start();
+        PrestoSparkTaskExecution taskExecution = new PrestoSparkTaskExecution(taskStateMachine, nonNativeTaskContext, localExecutionPlan, taskExecutor, TaskTestUtils.createTestSplitMonitor(), taskNotificationExecutor, scheduledExecutor, false);
+        taskExecution.start(ImmutableList.of(new TaskSource(TABLE_SCAN_NODE_ID, ImmutableSet.of(split1, split2, split3), true)));
+        assertEquals(nonNativeTaskContext.getPipelineContexts().get(0).getPipelineStats().getDrivers().size(), 3);
```

Comment:
Is this the only difference between the 2 tests (in addition to setting NATIVE_EXECUTION_ENABLED to true or false)? 

If that's the case, perhaps, combine these 2 tests into one to avoid so much copy paste.